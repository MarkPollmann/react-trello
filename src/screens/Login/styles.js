import styled from 'styled-components'

export const OuterWrapper = styled.div`
  display: flex;
  justify-content: center;
`

export const InnerWrapper = styled.div`
  width: 50%;
`

export const Input = styled.input`
  padding: 0.5rem;
  font-size: 16px;
  width: 100%;
  display: block;
  border-radius: 4px;
  border: 1px solid #ccc;
`

export const Warning = styled.p`
${'' /* color:red; */}
`

export const Button = styled.button`
  max-width: 150px;
  margin: 20px 0;
  padding: 12px 20px;
  border-style: none;
  border-radius: 5px;
  background-color: #08c;
  box-shadow: 0px 2px 2px rgba(0, 0, 0, 0.15);
  font-size: 17px;
  font-weight: 500;
  color: #fff;
  cursor: pointer;
  outline: none;
  -webkit-appearance: none;

  &:disabled {
    opacity: 0.5;
    cursor: not-allowed !important;
  }

  & + button {
    margin-left: 0.5rem;
  }

  ${'' /* button.outline {
    background-color: #eee;
    border: 1px solid #aaa;
    color: #555;
  } */};
`
