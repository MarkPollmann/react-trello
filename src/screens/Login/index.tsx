import { withFormik, Formik } from "formik";
import * as React from "react";
import { Link } from "react-router-dom";
import { Auth } from "aws-amplify";
import {
  Button,
  Divider,
  Form,
  Input,
  Label,
  Progress,
} from "semantic-ui-react";
import * as Yup from "yup";

import * as S from "./styles";

const Login = props => {
  const {

    // handleLogin
  } = props; 

  const handleLogin = async ({email, password}) => {
    try {
      const signInData = await Auth.signIn(email, password)
    //   if (signInData) {
    //     this.setState({isLoggedIn: true})
    //   }
      props.history.push('/app')
    } catch (error) {
      console.log(error)
      return error
    }
  }
  
  return (
    <S.OuterWrapper>
      <S.InnerWrapper>
        <h2>Log in to get access to your board</h2>
        <Formik
          initialValues={{ email: "testdrive@react-trello.com", password: "testdrive" }}
          onSubmit={async ({ email, password }, { setSubmitting }) => {
            setSubmitting(true);
            const error = await handleLogin({ email, password });
            if (error) {
              console.warn("error", error);
            }
            setSubmitting(false);
          }}

          // {
          //   handleSubmit: myHandleSubmit,
          //   mapPropsToValues: () => ({ email: 'testdrive@react-trello.com', password: 'testdrive' }),
          //   validationSchema: Yup.object().shape({
          //     email: Yup.string()
          //       .email()
          //       .required(),
          //     password: Yup.string().required(),
          //   }),
          // }
        >
          {({
            values,
            errors,
            touched,
            handleChange,
            handleBlur,
            handleReset,
            handleSubmit,
            isSubmitting,
            dirty,
          }) => {
            if (isSubmitting) {
              return (
                <div
                  style={{
                    alignItems: "center",
                    display: "flex",
                    flexDirection: "column",
                    height: "100%",
                    justifyContent: "center",
                  }}>
                  <div style={{ width: "50%" }}>
                    <Progress percent={100} indicating />
                  </div>
                  <h2>Looking for the keys ...</h2>
                </div>
              );
            }

            return (<Form onSubmit={handleSubmit}>
              <Form.Field>
                <Label pointing="below" htmlFor="email">
                  Your Email
                </Label>
                <Input
                  id="email"
                  type="text"
                  placeholder="Your email..."
                  value={values.email}
                  onChange={handleChange}
                  onBlur={handleBlur}
                />
                {errors.email && touched.email && (
                  <S.Warning>A valid email is required</S.Warning>
                )}
              </Form.Field>
              <Divider />
              <Form.Field>
                <Label pointing="below" htmlFor="password">
                  Your Password
                </Label>
                <Input
                  id="password"
                  type="password"
                  placeholder="Your password..."
                  value={values.password}
                  onChange={handleChange}
                  onBlur={handleBlur}
                />
                {errors.password && touched.password && (
                  <S.Warning>Enter your password</S.Warning>
                )}
              </Form.Field>
              <Button
                type="button"
                className="outline"
                onClick={handleReset}
                disabled={!dirty || isSubmitting}>
                Reset
              </Button>
              <Button type="submit" disabled={isSubmitting}>
                Submit
              </Button>
              {errors && (
                <h2>
                  No such user found. Maybe try{" "}
                  <Link to="/signup">signing up</Link>
                </h2>
              )}
            </Form>)
          }}
        </Formik>
        <br />
      </S.InnerWrapper>
    </S.OuterWrapper>
  );
};

// const myHandleSubmit = async (
//   { email, password },
//   { setSubmitting, props, setError },
// ) => {
//   setSubmitting(true)
//   const error = await props.handleLogin({ email, password })
//   if (error) {
//     setError('failed to login')
//   }
//   setSubmitting(false)
// }

export default Login;
