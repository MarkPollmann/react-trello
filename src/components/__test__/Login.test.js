import React from 'react'
import { render } from 'react-testing-library'
import 'jest-dom/extend-expect'
import Login from '../../screens/Login'

describe('<Login />', () => {
  it('matches snapshot', () => {
    const { container } = render(<Login />)
    expect(container).toMatchSnapshot()
  })

  it('renders a login box', () => {
    const { container } = render(<Login />)
    expect(container).toBeInTheDOM()
    expect(container).toHaveTextContent('email')
    expect(container).toHaveTextContent('password')
    expect(container).toHaveTextContent('Log in')
  })
})
