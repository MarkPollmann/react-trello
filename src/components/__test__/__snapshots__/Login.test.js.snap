// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Login /> matches snapshot 1`] = `
<div>
  <div
    class="sc-bdVaJa NFzwl"
  >
    <div
      class="sc-bwzfXH aKgAa"
    >
      <h2>
        Log in to get access to your board
      </h2>
      <form
        class="ui form"
      >
        <div
          class="field"
        >
          <div
            class="ui pointing below label"
            for="email"
          >
            Your Email
          </div>
          <div
            class="ui input"
          >
            <input
              id="email"
              placeholder="Your email..."
              type="text"
              value="testdrive@react-trello.com"
            />
          </div>
        </div>
        <div
          class="ui divider"
        />
        <div
          class="field"
        >
          <div
            class="ui pointing below label"
            for="password"
          >
            Your Password
          </div>
          <div
            class="ui input"
          >
            <input
              id="password"
              placeholder="Your password..."
              type="password"
              value="testdrive"
            />
          </div>
        </div>
        <button
          class="ui disabled button outline"
          disabled=""
          role="button"
          tabindex="-1"
          type="button"
        >
          Reset
        </button>
        <button
          class="ui button"
          role="button"
          type="submit"
        >
          Submit
        </button>
      </form>
      <br />
    </div>
  </div>
</div>
`;
